src/town_gui.cpp

```diff
@@ -1209,3 +1226,9 @@ void ShowFoundTownWindow()
 	if (_game_mode != GM_EDITOR && !Company::IsValidID(_local_company)) return;
 	AllocateWindowDescFront<FoundTownWindow>(&_found_town_desc, 0);
 }
+
+void InitializeTownGui()
+{
+	std::vector<TownID> townids;
+	_town_local_authority_kdtree.Build(townids.begin(), townids.end());
```

Comment:
this seems very suspect...
Not only is `townids` empty when this is passed in, but it goes out of scope and is destroyed when this function returns

Reply:
I have to make an empty vector to add identical iterators to `Build()`, as it does not work with null pointers. However as it does not look good for sure, I made a `Clear()` function in the kdtree instead, and it can be called without any parameters. It seems like a little change, but I can put it into a separate commit if needed.